### DISCUSSION TITLE
OSError: Timeout on getting final answer from server. Current state: DELIVERED

### DISCUSSION THREAD

**Discussion by martinorob** [bimmerconnected/bimmer_connected#190] [Q&A] [answered]
I can't receive the VEHICLE_FINDER because of timeout.

Every time I do

bimmerconnected vehiclefinder my@ema.il mypassword rest_of_world myvim

I receive 
DEBUG:bimmer_connected.remote_services:current state if remote service is: PENDING

36 time

then

DEBUG:bimmer_connected.remote_services:current state if remote service is: DELIVERED 

8 time

and then

OSError: Timeout on getting final answer from server. Current state: DELIVERED

and the command stops.

Is there any way to change timeout?

Thanks!!

**Comment by martinorob** [comment]
changed _POLLING_TIMEOUT = 60
to 
_POLLING_TIMEOUT = 600

now I have an
DEBUG:bimmer_connected.remote_services:current state if remote service is: EXECUTED
DEBUG:bimmer_connected.state:requesting new data from connected drive
DEBUG:bimmer_connected.account:Old token is still valid. Not getting a new one.
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): b2vapi.bmwgroup.com:443
DEBUG:urllib3.connectionpool:https://b2vapi.bmwgroup.com:443 "GET /webapi/v1/user/vehicles/myvim/status?deviceTime=2020-06-29T01%3A07%3A00&dlat=0.0&dlon=0.0 HTTP/1.1" 200 None
DEBUG:bimmer_connected.state:received new data from connected drive
ExecutionState.EXECUTED


but no data received.
How I can troubleshoot this?
Thanks

**Comment by rikroe** [comment]
The timeout issue is a good point, maybe we need to increase it. Could you please measure how long it takes for your vehicle to find it's position (i.e. from starting the service until it is successfully executed)?

`vehiclefinder` will not return any data, it will just update the coordinates in vehicle `status`.  So if you get the `status` after a `vehiclefinder` was successfull, you should see a difference (at least in the `updateTime`).

**Comment by martinorob** [comment]
So there is no way to get gps position?

**Comment by rikroe** [comment]
The new GPS position is visible when you do `bimmerconnected status my@ema.il mypassword rest_of_world`. 

If you're using HomeAssistant, the new GPS position will automatically be updated to your `device_tracker` entity every 5 minutes.

**Comment by martinorob** [comment]
No, I'm writing some script to tel Alexa where I parked my bmw last night 😊 
The VEHICLE_FINDER run for about 80 seconds. So I think that 120 second delay is ok.
 
ok in status response I have 

Vehicle status:
{
    "vin": "myvin,
    "updateTime": "2020-06-29T08:23:07+0000",
    "position": {
        "status": "TOO_FAR_AWAY"
    },
    "vehicleCountry": "IT"
}


I think this is because I have the 3km restriction and the VEHICLE_FINDER doesn't pass to bmw my current position.
Or I'm wrong? 

(with the BMW-Connect app on my iPhone I can find my car and it near me ).

Thanks

**Comment by rikroe** [comment]
`VEHICLE_FINDER` afaik does not have the 3km restriction. That purely handled on BMW server side.

To get the status including the position you need to give you current position (phone position in app context) to the cli status call (see the docs: https://bimmer-connected.readthedocs.io/en/latest/cli.html#status).

**Comment by martinorob** [comment]
Great! Thank you!
Last 2 questions:
1) is there any way to turn off Debug and have an exit code like on ExecutionState.EXECUTED echo EXECUTED?
2) how I can have a correct json output with Vehicle status information?
(without "Vehicle status:" line or "Found 1 vehicles: MYVEHICLE
VIN: MYVIN
Mileage: None
Vehicle properties:")

Thanks a lot!

**Comment by rikroe** [comment, marked as answer]
Sorry to leave you hanging there for so long. 
If you planning to build your own script to use with Alexa, I think it is best not to use the `bimmerconnected` commandline but to write your own little Python wrapper script. 

This way you could: 
- return the `ExecutionState`
- get the status as Python `dict` and format it in every way you like. 

To do so, just install `bimmer_connected` via pip and then import the relevant modules to your own script (`cli.py` should be a good reference there).

The 120 second timeout will be merged so the released version soon.
